proto/public/api.proto

```diff
@@ -133,5 +133,6 @@ service Api {
   rpc Stat(MetricRequest) returns (MetricResponse) {}
   rpc Version(Empty) returns (VersionInfo) {}
   rpc ListPods(Empty) returns (ListPodsResponse) {}
+  rpc SelfCheck(common.SelfCheckRequest) returns (common.SelfCheckResponse) {}
```

Comment:
I am thinking that every  component—maybe even proxies?—should be able to execute self-tests. I didn't want to mess too much with this until we can refactor the server code a bit but thought that adding this to a file different from the Conduit API would be better.

Now, I am not sure `common` is the best place, it seems like this namespace is becoming a dumping ground. Maybe a `healthcheck.proto`?

Reply:
Cool, this approach makes sense to me, and I agree that it would be good for all services to implement some sort of ping endpoint to check status (aside: maybe we should call this `rpc Ping`?).